maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkout/JGitSshCheckOutCommandTckTest.java

```diff
@@ -0,0 +1,149 @@
+        // accept all hosts
+        provider.registerCheckOutCommandCallback( new Consumer<JGitCheckOutCommand>()
+        {
+            @Override
+            public void accept( JGitCheckOutCommand command )
+            {
+                command.setSshSessionFactorySupplier( AcceptAllHostsSshdSessionFactory::new );
+            }
+
+        } );
+    }
+
+    private static final class AcceptAllHostsSshdSessionFactory extends ScmProviderAwareSshdSessionFactory
+    {
+        public AcceptAllHostsSshdSessionFactory( GitScmProviderRepository repo, Logger logger )
+        {
+            super( repo, logger );
+        }
+
+        @Override
+        protected ServerKeyDatabase createServerKeyDatabase( File homeDir, File sshDir )
+        {
+            return new OpenSshServerKeyDatabase( false, null )
+            {
+                @Override
+                public boolean accept( @NonNull String connectAddress,
+                        @NonNull InetSocketAddress remoteAddress,
+                        @NonNull PublicKey serverKey,
+                        @NonNull Configuration config, CredentialsProvider provider )
+                {
+                    return true;
+                }
+            };
+        }
+    }
+
+    @Override
+    protected void deleteDirectory( File directory )
+        throws IOException
+    {
+        if ( directory.exists() )
+        {
+            FileUtils.delete( directory, FileUtils.RECURSIVE | FileUtils.RETRY );
+        }
+    }
+
+    @Test
+    public void testCheckOutCommandTestWithOpenSSHPrivateKey()
+        throws Exception
+    {
+        configureKeypairFromResource( getScmRepository(), "sample_rsa", "mySecret");
+        super.testCheckOutCommandTest();
+    }
+
+    private void configureKeypairFromResource( ScmRepository repository, String resourceName, String passphrase )
+        throws IOException, GeneralSecurityException
+    {
+        // accept public key
+        try ( InputStream publicKeyInputStream = this.getClass().getResourceAsStream( "/ssh-keypairs/" + resourceName + ".pub"))
+        {
+            PublicKey publicKey = PublicKeyEntry.parsePublicKeyEntry( IOUtils.toString( publicKeyInputStream, StandardCharsets.US_ASCII ) ).resolvePublicKey( null, null, null );
+            acceptedPublicKeys.add( publicKey );
+        }
+        Path privateKeyFile = Files.createTempFile( "privateKey", null );
+        // private key into tmp file
+        try ( InputStream privateKeyInputStream = this.getClass().getResourceAsStream( "/ssh-keypairs/" + resourceName ))
+        {
+            Files.copy( privateKeyInputStream, privateKeyFile, StandardCopyOption.REPLACE_EXISTING );
+        }
+        // configure provider repository with private key details
+        ScmProviderRepositoryWithHost providerRepository = 
+                        ScmProviderRepositoryWithHost.class.cast( repository.getProviderRepository() );
+        if ( passphrase != null )
```

Comment:
Why test, set it right away, if it is internally, `null` then the result will be the same, no?

Reply:
Right, simplified in https://github.com/apache/maven-scm/pull/155/commits/9f86604ea544ad802dffcd48e5c381facebb9269.